### PR TITLE
Cancel pipeline run api

### PR DESCRIPTION
**Pull request by milantomic6**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/datafactory/resource-manager/Microsoft.DataFactory/2017-09-01-preview/datafactory.json [hunk 1]
```diff
@@ -1770,6 +1770,48 @@
...
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataFactory/factories/{factoryName}/cancelpipelinerun/{runId}": {
+      "post": {
+        "operationId": "Factories_CancelPipelineRun",
+        "x-ms-examples": {
+          "Factories_CancelPipelineRun": {
+            "$ref": "./examples/Factories_CancelPipelineRun.json"
+          }
+        },
+        "description": "Cancel pipeline run by its run ID.",
```
Comment (bgold09) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/2017-09-01-preview/datafactory.json]: "Cancel a pipeline run..."

File: specification/datafactory/resource-manager/Microsoft.DataFactory/2017-09-01-preview/datafactory.json [hunk 2]
```diff
@@ -1770,6 +1770,48 @@
...
+            "$ref": "#/parameters/factoryName"
+          },
+          {
+            "$ref": "#/parameters/runId"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
```
Comment (bgold09) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/2017-09-01-preview/datafactory.json]: What happens if the pipeline run has already finished?

File: specification/datafactory/resource-manager/Microsoft.DataFactory/2017-09-01-preview/datafactory.json [hunk 3]
```diff
@@ -1770,6 +1770,48 @@
...
+          "200": {
```
Comment (bgold09) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/2017-09-01-preview/datafactory.json]: When the user gets an `OK` response, does that mean the run has been cancelled? Or does it mean that we've started the process of cancelling it?

File: specification/datafactory/resource-manager/Microsoft.DataFactory/2017-09-01-preview/datafactory.json (hunk 2 above)
Comment (milantomic6) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/2017-09-01-preview/datafactory.json]: If pipeline run is in terminal state we return bad request.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/2017-09-01-preview/datafactory.json (hunk 3 above)
Comment (milantomic6) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/2017-09-01-preview/datafactory.json]: Ok means it is fully cancelled.

Comment (sergey-shandar) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/2017-09-01-preview/datafactory.json]: @milantomic6 could you add to description.

**Review by sergey-shandar — CHANGES_REQUESTED** [review]
Please, address comments from @bgold09

**Review by sergey-shandar — APPROVED** [review]

**Pull request merged**